### PR TITLE
Cleanup pytest targets

### PR DESCRIPTION
**Pull request by charlieparkes**

### CODE REVIEW

File: modules/pytest/Makefile [hunk 1]
```diff
@@ -1,14 +1,50 @@
...
+
+.PHONY: pytest/post-build
+pytest/post-build: ; @$(MAKE) pytest/m/"postbuild"
+
+####
+# Deprecated targets.
+####
+
+define pytest-deprecated
+	@echo "WARNING: This target is being deprecated. Please switch from '$(1)' to '$(2)'"
```
Comment (chavlin) [inline modules/pytest/Makefile]: nit: `deprecation` already refers to the process of future replacement, so I think it's better to say `This target is deprecated.`

**Review by chavlin — APPROVED** [review]

File: modules/pytest/Makefile (hunk 1 above)
Comment (charlieparkes) [reply modules/pytest/Makefile]: Fixed

File: modules/pytest/Makefile [hunk 2]
```diff
@@ -1,14 +1,50 @@
-pytest: pytest/test
+PYTEST = $(PIPENV_RUN) pytest
+
+.PHONY: pytest/init
+pytest/init: pipenv reports/
+
 .PHONY: pytest
+## Run Python tests with pytest
+pytest: pytest/init ; $(PYTEST)
```
Comment (jtdoepke) [inline modules/pytest/Makefile]: Can you put the commands on a separate line?

```Makefile
pytest: pytest/init
    $(PYTEST)
```
I think it's more readable that way; you can tell at a glance if something is a dependency target or a command.

**Review by chavlin — APPROVED** [review]

File: modules/pytest/Makefile (hunk 2 above)
Comment (charlieparkes) [reply modules/pytest/Makefile]: Personally, I like shortening, but I'm not that attached. Change made.

**Review by jtdoepke — APPROVED** [review]

**Pull request merged**